File: src/main/java/com/innatour/toursmanager/dto/JwtResponse.java
package com.innatour.toursmanager.dto;

/**
 * JWT Response DTO
 * Returned after successful login
 * Contains the JWT token that the frontend will use for authenticated requests
 */
public class JwtResponse {
    
    private String token;
    private String type = "Bearer";  // Token type (standard for JWT)
    private String email;
    private String firstName;
    private String lastName;
    
    public JwtResponse(String token, String email, String firstName, String lastName) {
        this.token = token;
        this.email = email;
        this.firstName = firstName;
        this.lastName = lastName;
    }
    
    public String getToken() {
        return token;
    }
    
    public void setToken(String token) {
        this.token = token;
    }
    
    public String getType() {
        return type;
    }
    
    public void setType(String type) {
        this.type = type;
    }
    
    public String getEmail() {
        return email;
    }
    
    public void setEmail(String email) {
        this.email = email;
    }
    
    public String getFirstName() {
        return firstName;
    }
    
    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }
    
    public String getLastName() {
        return lastName;
    }
    
    public void setLastName(String lastName) {
        this.lastName = lastName;
    }
}

File: dotnet-api/ToursManager.Api/DTOs/JwtResponse.cs
namespace ToursManager.Api.DTOs;

/// <summary>
/// JWT Response DTO - returned after successful login.
/// Contains the JWT token that the frontend will use for authenticated requests.
/// </summary>
public class JwtResponse
{
    public string Token { get; set; } = string.Empty;
    
    /// <summary>
    /// Token type (standard for JWT)
    /// </summary>
    public string Type { get; set; } = "Bearer";
    
    public string Email { get; set; } = string.Empty;
    
    public string FirstName { get; set; } = string.Empty;
    
    public string LastName { get; set; } = string.Empty;
    
    public JwtResponse(string token, string email, string firstName, string lastName)
    {
        Token = token;
        Email = email;
        FirstName = firstName;
        LastName = lastName;
    }
}
